src/ds5/ds5-color.h

```diff
@@ -52,8 +54,33 @@ namespace librealsense
         stream_profiles init_stream_profiles() override;
         processing_blocks get_recommended_processing_blocks() const override;
 
-    private:
+    protected:
         const ds5_color* _owner;
     };
 
+    // The class allows to invoke calibraiton routine on RGB sensor,
+    // and also retrieve the updated calibration data on the fly
+    // Specifically for D400 the calibration adjustments are generated in FW
+    // and retrieved on demand when a certain (thermal) trigger is invoked
+    class ds5_recalibrable_color_sensor : public ds5_color_sensor,
+                                          public calibrated_sensor
```

Comment:
I don't think `calibrated_sensor` implementation is right here.
I think what you want is `d455_color_sensor` that inherits from `ds5_color_sensor` impl and implements `ds5_recalibrable_color_sensor` abstract interface.

Reply:
Refactored